src/Microsoft.VisualStudio.Validation/Assumes.cs

```diff
@@ -194,7 +196,7 @@ public static Exception NotReachable()
         /// </summary>
         /// <typeparam name="T">The interface of the imported part.</typeparam>
         [DebuggerStepThrough]
-        public static void Present<T>(T component)
+        public static void Present<T>([NotNull]T component)
```

Comment:
If you decide against removing `?` from all parameters then we should allow a nullable type to flow in here for consistency.

```suggestion
        public static void Present<T>([MaybeNull, NotNull]T component)
```

Reply:
This `T` can't be `T?` because there is no `T: class` type constraint. So it's always `T?`. Furthermore, I don't think `MaybeNull` is appropriate here because we will never return from this method if `component == null` as we throw in that case. 
What would `[MaybeNull, NotNull]` even mean anyway? Those two together seem to contradict.

Reply:
Looking at this again my intent was to say `[AllowNull, NotNull] T`, intending that to be functionally equivalent to `[NotNull] T?` but for unconstrained generic `T`.

If we're not allowing nullable types to flow into these methods then the point is moot.

Reply:
From my testing, `[NotNull]` doesn't cause the compiler to forbid callers to pass in null -- but it does effectively convince the compiler that by the time the method returns that the value must not be null.
So with `[NotNull]T` where `T:class` is *not* a type constraint, callers see `T?` in the signature and can pass in null with no warning. After this method call because of `[NotNull]` being present, the compiler lets them dereference `T` with no warning. 
So I'm not sure what effect you're going for that adding `[AllowNull]` would give. We implicitly allow null already.